pkg/dns/ibm/common.go

```diff
@@ -0,0 +1,40 @@
+package common
+
+import (
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	iov1 "github.com/openshift/api/operatoringress/v1"
+)
+
+// Common configuration of ibm providers
+// InstanceID is GUID in case of dns svcs and CRN in case of cis
+type Config struct {
+	APIKey     string
+	InstanceID string
+	UserAgent  string
+	Zones      []string
```

Comment:
This would be more Go idiomatic:
```suggestion
// Config holds common configuration of IBM DNS providers.
type Config struct {
	// APIKey is the IBM Cloud API key that the provider will use.
	APIKey     string
	// InstanceID is GUID in case of dns svcs and CRN in case of cis.
	InstanceID string
	// UserAgent is the user-agent identifier that the client will use in
	// HTTP requests to the IBM Cloud API.
	UserAgent  string
	// Zones is a list of DNS zones in which the DNS provider manages DNS records.
	Zones      []string
```

Reply:
Thank you. Applied your suggestion.